src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -1,169 +1,115 @@
-
-  const [pictureUrl, setPictureUrl] = useState("");
-
-  const randomAvatarUrl = (id: string) =>
-    "/avatars/" +
-    RANDOM_AVATARS[Math.floor(id?.charCodeAt(0) % RANDOM_AVATARS.length)];
-
-  const avatarUrl = useCallback(
-    (id: string, anonMode?: boolean, pictureUrl?: string) => {
-      if (anonMode || !id) {
-        return setPictureUrl(DEFAULT_PROFILE_IMAGE);
-      }
-
-      if (!miniAvatars && pictureUrl) {
-        return setPictureUrl(pictureUrl);
-      }
-
-      if (miniAvatars) {
-        return setPictureUrl(randomAvatarUrl(id));
-      }
-
-      return setPictureUrl(DEFAULT_PROFILE_IMAGE);
-    },
-    [miniAvatars]
-  );
+  const userId = user?.id;
 
-  useEffect(() => {
-    avatarUrl(user.id, user.anonMode, user.pictureUrl);
-  }, [avatarUrl, user.anonMode, user.id, user.pictureUrl]);
-
-  const venueId = useVenueId();
-  const reactions = useReactions(venueId);
-
-  const typedReaction = reactions ?? [];
-
-  const messagesToBand = typedReaction.find(
-    (r) => r.reaction === "messageToTheBand" && r.created_by === user.id
-  ) as MessageToTheBandReaction | undefined;
+  const { openUserProfileModal } = useProfileModalControls();
 
-  const imageErrorHandler = useCallback(
-    (
-      event: HTMLImageElement | React.SyntheticEvent<HTMLImageElement, Event>
-    ) => {
-      const randomAvatar = randomAvatarUrl(user.id);
-      setPictureUrl(randomAvatar);
+  const openProfileModal = useCallback(() => openUserProfileModal(user), [
+    openUserProfileModal,
+    user,
+  ]);
 
-      (event as HTMLImageElement).onerror = null;
-      (event as HTMLImageElement).src = randomAvatar;
-    },
-    [user.id]
+  // const pictureUrl = getAvatarUrl({ user, miniAvatars });
+
+  // @debt useImage tries to load the images twice, which is made worse by us not caching images retrieved from firebase,
+  //  it's only used to handle the edgecase of showing a default when images are missing. Can we live without it?
+  // const { loadedImageUrl: pictureUrl } = useImage({
+  //   src: avatarUrl({ user, miniAvatars }),
+  //   fallbackSrc: () => randomAvatarUrl(user.id),
+  // });
+
+  // const containerVars = useCss({
+  //   "--user-profile-picture-avatar-url": pictureUrl
+  //     ? `url(${pictureUrl})`
+  //     : `url(${DEFAULT_PROFILE_IMAGE})`,
+  // });
+
+  const containerClasses = classNames(
+    "UserProfilePicture",
+    containerClassName
+    // containerVars
   );
 
-  return useMemo(() => {
-    return (
-      <S.Container style={{ ...containerStyle }}>
-        {/* Hidden image, used to handle error if image is not loaded */}
-        <img
-          src={pictureUrl}
-          onError={(event) => imageErrorHandler(event)}
-          hidden
-          style={{ display: "none" }}
-          alt={user.anonMode ? DEFAULT_PARTY_NAME : user.partyName}
+  // const userDisplayName: string = user?.anonMode
+  //   ? DEFAULT_PARTY_NAME
+  //   : user?.partyName ?? DEFAULT_PARTY_NAME;
```

Comment:
Deleted/comment?

Reply:
Added `@debt` comments to explain the commented code blocks in https://github.com/sparkletown/sparkle/pull/1323/commits/5f2acbd92b4f96c175bf39ab1ffc0c0bb7d92624